app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -142,15 +142,25 @@ def data(payload)
     end
   end
 
-  def cleanup(*_args)
+  def cleanup(*args)
+    begin
+      process_cleanup
+    rescue Exception => e
+      _log.error("Exception during cleanup: #{e}")
+    end
+
+    args.empty? ? queue_signal(:finish) : process_abort(*args)
```

Comment:
@moolitayer I think you could have just added to the existing `cleanup` code:

    def cleanup(*_args)
      ...
    ensure
      args.empty? ? queue_signal(:finish) : process_abort(*args)
    end